polling_stations/apps/data_importers/import_script.py

```diff
@@ -20,27 +43,161 @@ def importer_class(self):
         return classes[self.ems]
 
     @property
-    def csv_encoding_string(self):
-        if self.encoding == "utf-8":
-            return ""
-
-        return f'\n    csv_encoding = "{self.encoding}"'
+    def delimiter(self):
+        return "\t" if self.addresses_name[-3:] in ("tsv", "TSV") else ","
 
     @property
-    def csv_delimiter_string(self):
-        if self.addresses_name[-3:] in ("tsv", "TSV"):
-            return '\n    csv_delimiter = "\\t"'
+    def script(self):
+        try:
+            script = self._get_updated_existing_script()
+        except Exception:
+            logger.exception(
+                "Exception modifying existing script; returning simple script instead."
+            )
 
-        return ""
+            script = textwrap.dedent(
+                f"""\
+                from data_importers.management.commands import {self.importer_class}
 
-    @property
-    def script(self):
-        return f"""from data_importers.management.commands import {self.importer_class}
 
+                class Command({self.importer_class}):
+                    council_id = "{self.council_id}"
+                    addresses_name = "{self.addresses_name}"
+                    stations_name = "{self.stations_name}"
+                    elections = {self.elections}{self.csv_encoding_string}{self.csv_delimiter_string}
+            """
+            )
+
+        # Run the updated script through black before returning it
+        return black.format_str(script, mode=black.Mode())
+
+    def _get_updated_existing_script(self):
+        """Update an existing script with the new Command parameters.
+
+        This uses the `ast` module to parse the existing script, and then:
+
+        * ensures that the Command has the right base class
+        * adds, removes and updates assignment statements in the Command as necessary
+
+        It has a couple of limitations:
+
+        * Multiple assignments are ignored, so a new assignment will be added later on.
+          This can be tidied up by hand.
+
+        * It might break an "import [...].commands from [...]" line if the original line
+          didn't import only the previous command base class.
```

Comment:
so if the line was `from data_importers.management.commands import BaseHalaroseCsvImporter, AnotherImporter` it'll replace both imports with the one expected importer class name, even if it was still used somewhere else in the script.

In practice, I don't think this would be an issue, and would be easily spotted if it did happen.